### ISSUE TITLE
Connection leak when sender.Recover() is called across multiple goroutines

### ISSUE THREAD

**Issue by jhendrixMSFT** [Azure/azure-event-hubs-go#215]
In the recovery mechanism [here](https://github.com/Azure/azure-event-hubs-go/blob/717fe444f26da332e23657da0a9e5de441bb249f/sender.go#L172), the recreation of the session and link is not goroutine-safe and can lead to leaked connections.

**Comment by jhendrixMSFT** [comment]
```
==================
WARNING: DATA RACE
Write at 0x00c0001743c8 by goroutine 84:
  github.com/Azure/azure-event-hubs-go.(*sender).newSessionAndLink()
      C:/git/Azure/azure-event-hubs-go/sender.go:231 +0x17b
  github.com/Azure/azure-event-hubs-go.(*sender).Recover()
      C:/git/Azure/azure-event-hubs-go/sender.go:89 +0x2a7
  github.com/Azure/azure-event-hubs-go.(*Hub).Recover()
      C:/git/Azure/azure-event-hubs-go/hub.go:581 +0x8c
  main.main.func1()
      C:/git/jhendrixMSFT/eventhubsrepro/main.go:26 +0x8d

Previous write at 0x00c0001743c8 by goroutine 73:
  github.com/Azure/azure-event-hubs-go.(*sender).newSessionAndLink()
      C:/git/Azure/azure-event-hubs-go/sender.go:231 +0x17b
  github.com/Azure/azure-event-hubs-go.(*sender).Recover()
      C:/git/Azure/azure-event-hubs-go/sender.go:89 +0x2a7
  github.com/Azure/azure-event-hubs-go.(*Hub).Recover()
      C:/git/Azure/azure-event-hubs-go/hub.go:581 +0x8c
  main.main.func1()
      C:/git/jhendrixMSFT/eventhubsrepro/main.go:26 +0x8d
```

**Comment by jhendrixMSFT** [comment]
Fixed in v3.3.10
